client/my-sites/pages/main.jsx

```diff
@@ -81,24 +81,10 @@ class PagesMain extends React.Component {
 			scheduled: translate( 'Scheduled', { context: 'Filter label for pages list' } ),
 			trashed: translate( 'Trashed', { context: 'Filter label for pages list' } ),
 		};
-		const searchStrings = {
-			published: translate( 'Search Published…', {
-				context: 'Search placeholder for pages list',
-				textOnly: true,
-			} ),
-			drafts: translate( 'Search Drafts…', {
-				context: 'Search placeholder for pages list',
-				textOnly: true,
-			} ),
-			scheduled: translate( 'Search Scheduled…', {
-				context: 'Search placeholder for pages list',
-				textOnly: true,
-			} ),
-			trashed: translate( 'Search Trashed…', {
-				context: 'Search placeholder for pages list',
-				textOnly: true,
-			} ),
-		};
+		const searchPlaceholder = translate( 'Search Pages…', {
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **148** times:
`translate( 'Search Pages' )` *ES Score: 7*


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

Reply:
@arunsathiya Could you explain what it means by "translator comment"? I am not familiar with the way you use translation on this repo, so I am not sure what is the best way forward to fix this.

Also, I looked on it was done on `Posts`, as it is basically the same translation ("Search posts..."), and I found this:

https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/post-type-filter/index.jsx#L208

I am not sure it is the best way to translate plural...

Reply:
Hey hey @millerf !

You almost always want to use a comment rather than context. This can be done in a note above the line with the string, like so:

`/* translators: %1$s is the post date, %2$s is the post author name */`

Comments add information for translators when a string uses variables and needs to know what type of information those variables might contain. I don't think you need to add a comment for "Search pages" because it's self explanatory.

Context is used to note when a string might need different translations depending on their location or reference to another element. For example, the string "Add New" could apply to Posts, Pages, Media, etc. In gendered languages, the context might change depending on the gender of the word "Add New" applies to.

Hope this makes sense!

Reply:
Technically I think there's an argument for using context here ("Search Pages" and the existing strings it is replacing can have two meanings depending on whether "Search" is a verb or adjective) but I guess it's better not to use a context if we don't have a specific need to distinguish them.

>Also, I looked on it was done on Posts, as it is basically the same translation ("Search posts..."), and I found this:
>
>https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/post-type-filter/index.jsx#L208
>
>I am not sure it is the best way to translate plural...

Good idea to check for that!  Although not critical, I think it might actually be a good idea to reuse that string if possible (so as not to introduce any new untranslated strings).  For reference, it's:

```
placeholder={ this.props.translate( 'Search %(postTypes)s…', {
	args: {
		postTypes: typeLabel,
	},
} ) }
```

I think that `typeLabel` already takes care of the pluralization, so it should be fine to use.  (Was your "I am not sure it is the best way" comment based on the "s" in `%(postTypes)s`?  Note that that isn't part of the translated text itself -- it just indicates that the substituted variable is a string; see http://www.diveintojavascript.com/projects/javascript-sprintf.)

Reply:
@sixhours Thanks a lot for the clarification! Now the 18n warning makes more sense 👍 

@DavidRothstein Never really looked or used `sprintf` in javascript before... And I guess I am used to work with bad devs who pluralize strings that way 🤦‍♂ 
That's really interesting.

I started to look into `typeLabel` this morning but I couldn't make it work with Pages. Although both of the headers work pretty much the same way (for posts and pages), they are build with a different mindset and I struggled a bit. I might give it another shot later.
Although, if I am not mistaken, typeLabel always returns "Posts" in this case. I don't think it is used anywhere else...

Reply:
I stand corrected. After digging some more, I found out `PostTypeFilter` is also used in `types`, but the typeLabel, which comes from the `query`, is the same...

Reply:
Great!  Looks like you're doing it like this now:

```
placeholder={ this.props.translate( 'Search %(postTypes)s…', {
  args: {
    postTypes: translate( 'Pages' ),
  },
} ) }
```

Sorry I wasn't too clear in my earlier comment -- I was actually thinking we'd get the "Pages" dynamically from the actual post type data (the same way the existing code you pointed to does it), rather than hardcoding it as "Pages".

I guess the way you did it works OK too (and is similar); it's just that if we're going to hardcode it in the end, maybe it's more straightforward (and better for translation) to just go back and have it be "Search Pages" after all -- and then live with the fact that it will need to be retranslated.  Not sure; maybe someone else has more experience with this.

Reply:
Either way. I reused a translation used earlier in the file, so I don't understand what you mean by 'hardcoded':
https://github.com/Automattic/wp-calypso/pull/40464/files#diff-1f00fdd573a2d25ec14b5bab3a884029L109

@DavidRothstein if you can shed some light here: I gave a shot at getting 'Pages' dynamically [on a new branch](https://github.com/millerf/wp-calypso/compare/fix%2Frebased-search-pages-across-statuses...millerf:try%2Fuse-label-to-search-page?diff=split) . It is not that straightforward as `query` (and `query.type`) is defined in a child of `PagesMain`, so I had to extract it in order not to hardcode. We could just pass "page" (the `query.type` value) as a prop, or use it as a constant. Not sure.

Reply:
@millerf there are a few labels that are powered by the API, like "Add new post". I really don't think it's necessary to power this with the API value though.

We can probably simplify this down to `translate( 'Search Pages' )` so this PR can land first with useful translations, then follow up with a copy only PR for the placeholder to add in the ellipsis if we think it's important.

Reply:
@gwwar Ok. I will push push a commit right now then, so you can land this one quickly.
Regarding translations in general, is there a list of the translations provided by the API somewhere?

Reply:
A quick question though. Right now it is translated like this: 
```
placeholder={ this.props.translate( 'Search %(postTypes)s…', {
  args: {
    postTypes: translate( 'Pages' ),
  },
} ) }
```

We know "Pages" is already translated, and "Search {postType}..." as well (from the `posts` pages), so there shouldn't be any problem for the translation as it is right now... I just don't see what the problem is...

Reply:
Usually args is populated by non-static data, or is an html fragment, so just a little unusual usage here.

Reply:
>I gave a shot at getting 'Pages' dynamically on a new branch

Looks promising!  I think it's probably worth pursuing further after this one gets in.  In general, I think it might be nice to just have a helper function that returns "Search [type]..." as the search placeholder for any post type (and then use that everywhere for consistency).  And it looks like you're about 80% of the way there.

Reply:
"Search Pages" and "Search Posts" are actually already translated by the API...

![Screenshot 2020-03-28 at 11 58 22](https://user-images.githubusercontent.com/790558/77821543-82a99a00-70eb-11ea-9a49-fa04aa77afe6.png)

@DavidRothstein I updated [this branch](https://github.com/millerf/wp-calypso/compare/fix%2Frebased-search-pages-across-statuses...millerf:try%2Fuse-label-to-search-page?diff=split) with the changes you proposed. It now retrieves the translation directly from `postType.labels.search_items`. I am still not sure on how to introduce the ellipsis.

```

	placeholder={ getPostTypeLabel( state, siteId, query.type, 'search_items' ) + '...' }

```

seems wrong, but translating the only ellipsis seems even worse...

```
	placeholder={ this.props.translate( '(searchItems)s…', {
		args: {
			searchItems: getPostTypeLabel( state, siteId, query.type, 'search_items' ),
		},
	} ) }
```

What do you think?

Reply:
Reusing the translations provided by the API seem a good idea. Likely these are defined by WordPress Core and will be already using a case sentence format when https://core.trac.wordpress.org/ticket/49616 is fixed. In the meantime, we could use the same JS workaround we used in https://github.com/Automattic/wp-calypso/pull/40256.

In regards to the ellipsis, I prefer the first option. It doesn't look that bad when using template literals:

```
const postTypeLabel = getPostTypeLabel( state, siteId, query.type, 'search_items' );

placeholder={ `${ postTypeLabel }…` }
```

Reply:
If you look at the branch, before using the API translation, I had a commit using the postTypeLabel. I just have to revert the last commit.
Should I create an issue for it? Or add it to #40553?

Reply:
> I just have to revert the last commit.

Yup, but you also need to remove the translation. `postTypeLabel` + appending ellipsis should be enough. I'll be easier to leave feedback once you have the PR open.

No need to create a new issue, #40553 is a good fit for this.